stablehlo/tools/StablehloInterpreterMain.cpp

```diff
@@ -27,41 +31,92 @@ limitations under the License.
 
 namespace mlir {
 
+namespace {
+
+stablehlo::Tensor makeBooleanTensor(MLIRContext *context, bool value) {
+  auto builder = Builder(context);
+  auto type = RankedTensorType::get({}, builder.getI1Type());
+  auto res = DenseElementsAttr::get(type, builder.getBoolAttr(true));
+  return stablehlo::makeTensor(res);
+}
+
+llvm::Error evalCustomCallCheckEq(stablehlo::CustomCallOp op,
+                                  stablehlo::Scope &scope,
+                                  llvm::StringRef funcName) {
+  if (op->getNumOperands() != 2) {
+    return stablehlo::invalidArgument("Unsupported op: %s",
+                                      debugString(op).c_str());
+  }
+  stablehlo::Tensor actualResult = scope.find(op->getOperands())[0];
+  auto expectedOp =
+      dyn_cast<stablehlo::ConstantOp>(op->getOperands()[1].getDefiningOp());
+
+  // TODO: Need to handle complex types.
+  bool isInt = expectedOp.getType().getElementType().isa<IntegerType>();
```

Comment:
Only for ints and bools `check.eq` is relevant. For others, `check.almost_eq` makes sense. So, maybe we do not have to handle complex separetly.